cranelift-frontend/src/switch.rs

```diff
@@ -129,21 +169,38 @@ impl Switch {
     ) {
         for ContiguousCaseRange { first_index, ebbs } in contiguous_case_ranges.into_iter().rev() {
             if ebbs.len() == 1 {
-                let is_good_val = bx.ins().icmp_imm(IntCC::Equal, val, first_index as i64);
-                bx.ins().brnz(is_good_val, ebbs[0], &[]);
+                if first_index == 0 {
```

Comment:
(light suggestion) There might be a way to reduce the number of code indents here by using `match (ebbs.len(), first_index)` with 4 cases.